### PR TITLE
DI/ViewModel

### PR DESCRIPTION
**Pull request by JJSwigut**
So I think I did everything correctly and honestly, Hilt was way easier than I expected! I know it's a super simple app but I can definitely see how it would make a much bigger app a lot easier. I decided to make a repository but I think it was a bit much for this app. I also think I could have done with one viewModel instead of one for each fragment, but I wanted your thoughts on it. What else would I have to do to this app(besides a lot more features) for this to look like a professional app? Are the packages structured right? Are things named correctly and the code generally readable? Thanks!

### CODE REVIEW

**Comment by j-roskopf** [comment]
Hilt definitely simplified things a ton for dagger / DI (which is a good thing!)

even though one ViewModel to cover 2 fragments might seem good, it's generally not. You want to separate those 2 fragments out so having one VM per fragment / activity is generally a pretty good way to go

File: app/src/main/java/com/jjswigut/matters/ui/matter/EditMatterFragmentViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.jjswigut.matters.database.Matter
+import com.jjswigut.matters.database.MatterRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class EditMatterFragmentViewModel @ViewModelInject constructor(
+    private val repo: MatterRepository,
+    @Assisted private val savedStateHandle: SavedStateHandle
```
Comment (j-roskopf) [inline app/src/main/java/com/jjswigut/matters/ui/matter/EditMatterFragmentViewModel.kt]: if you aren't using the saved state you can just remove it

File: app/src/main/java/com/jjswigut/matters/ui/matterlist/MatterListFragmentViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+) : ViewModel() {
+
+    val allMatters: LiveData<List<Matter>> = repo.allMatters
+
+
+    fun insert(matter: Matter) = viewModelScope.launch(Dispatchers.IO) {
+        repo.insert(matter)
+    }
+
+    fun delete(matter: Matter) = viewModelScope.launch(Dispatchers.IO) {
```
Comment (j-roskopf) [inline app/src/main/java/com/jjswigut/matters/ui/matterlist/MatterListFragmentViewModel.kt]: are you using any of these CRUD functions on the main page?

**Comment by j-roskopf** [comment]
Hilt definitely simplified things a ton for dagger / DI (which is a good thing!)

even though one ViewModel to cover 2 fragments might seem good, it's generally not. You want to separate those 2 fragments out so having one VM per fragment / activity is generally a pretty good way to go

I can take a look after it's all done to comment on things like package structure, but once you having unit and instrumentation tests running, i think you're looking pretty close to having a fully functioning app!

It's pretty hard to demonstrate on a smaller project like this, but one thing to consider for the next project is to do some more modularization of the app. It'll be a little easier in the next app when you have a networking stack, but otherwise i think it's looking good!

File: app/src/main/java/com/jjswigut/matters/ui/matterlist/MatterListFragmentViewModel.kt (hunk 2 above)
Comment (JJSwigut) [reply app/src/main/java/com/jjswigut/matters/ui/matterlist/MatterListFragmentViewModel.kt]: nope, I just copy and pasted the viewmodel. which was part of the reason I thought it wasn't necessary to have 2. I get what you mean by keeping things separate though!

**Pull request merged**